src/critnib/critnib.c

```diff
@@ -176,37 +182,68 @@ static inline unsigned slice_index(word key, sh_t shift) {
  * critnib_new -- allocates a new critnib structure
  */
 struct critnib *critnib_new(void) {
-    struct critnib *c = Zalloc(sizeof(struct critnib));
-    if (!c) {
+    umf_ba_linear_pool_t *pool_linear =
+        umf_ba_linear_create(0 /* minimal pool size */);
+    if (!pool_linear) {
         return NULL;
     }
 
-    c->mutex = util_mutex_create();
+    struct critnib *c =
+        umf_ba_linear_alloc(pool_linear, sizeof(struct critnib));
+    if (!c) {
+        goto err_destroy_pool_linear;
+    }
+
+    c->pool_linear = pool_linear;
+
+    void *mutex_ptr = umf_ba_linear_alloc(pool_linear, util_mutex_get_size());
```

Comment:
nit: since you are modyfing this part anyway, you can actually just refactor this part and store the mutex directly in the critnib stuct (just use util_mutex_init, without allocation). This way we could also get rid of the util_mutex_* functions that allocate/free memory and only leave (util_mutex_destroy and util_mutex_create)

Reply:
Done